### PR TITLE
examples: standardize docker-compose version and yaml extension

### PR DESCRIPTION
**Pull request by cpaika**
Description:
Following @moderation comment from [this pr](https://github.com/envoyproxy/envoy/pull/6574) to upgrade the docker-compose version for every example.  
Standardizing on .yaml extension per their comment
Changing the build script name in grpc-bridge-example to [match the documentation](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/grpc_bridge#building-the-go-service)

Risk Level: Low
Testing: Manual - brought up every example after upgrading the docker-compose version
Docs Changes: Yes - just matched the new file extensions
Release Notes: No

### CODE REVIEW

File: examples/cors/backend/docker-compose.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: '2'
+version: '3.7'
```
Comment (moderation) [inline examples/cors/backend/docker-compose.yaml]: Nit. I didn't mention this on my comment on the prior issue but for consistency with quoting in YAML files can you also change the quoting here from single quote to double quote. ie "3.7". Same for the other occurrences.

Comment (cpaika) [reply examples/cors/backend/docker-compose.yaml]: Gotcha, will update thanks 👍

**Comment by cpaika** [comment]
@moderation Makes sense - good memory haha. Will update the docs and revert the name change.  Note to self always check git blame before changing a line...

File: docs/root/start/sandboxes/grpc_bridge.rst [hunk 2]
```diff
@@ -24,9 +24,9 @@ To build the Go gRPC service run::
   $ pwd
   envoy/examples/grpc-bridge
   $ script/bootstrap
```
Comment (moderation) [inline docs/root/start/sandboxes/grpc_bridge.rst]: I suggest your rename `bootstrap` to `bootstrap.sh` here for consistency and updating and update the doco. Similarly rename `grpc_start` to `grpc_start.sh` and update the Dockerfile. This way all shell scripts in the directly will have .sh extensions.

Comment (cpaika) [reply docs/root/start/sandboxes/grpc_bridge.rst]: Good point, will do 👍

**Comment by cpaika** [comment]
Updated to follow comments @moderation

**Comment by cpaika** [comment]
Could I also get a retest?
The test that failed was `//test/common/event:dispatcher_impl_test` which seems unrelated

**Comment by moderation** [comment]
@cpaika You can kick off your own retest - see https://github.com/envoyproxy/envoy/blob/master/source/docs/repokitteh.md#circleci-retest

LGTM otherwise. Thanks

**Comment by cpaika** [comment]
@moderation Thanks for the reviews on this!  I'm looking forward to getting involved so if you know of anything that is a good beginner issue and need someone to work on please let me know.  Otherwise I'll start looking through the existing issues for something that hasn't been grabbed.

**Pull request merged**